components/proxy/src/main/java/com/hotels/styx/routing/handlers/ProxyToBackend.java

```diff
@@ -99,14 +102,33 @@ public HttpHandler2 build(List<String> parents, RouteHandlerFactory builder, Rou
                     .longFormat(longFormat)
                     .build();
 
-            OriginsInventory inventory = new OriginsInventory.Builder(backendService)
-                    .version(environment.buildInfo().releaseVersion())
+
+            ConnectionPool.Factory connectionPoolFactory = connectionPoolFactory(
+                    connectionFactory,
+                    backendService.connectionPoolConfig(),
+                    environment.metricRegistry()
+            );
+
+            OriginsInventory inventory = new OriginsInventory.Builder(backendService.id())
                     .eventBus(environment.eventBus())
                     .metricsRegistry(environment.metricRegistry())
-                    .connectionFactory(connectionFactory)
-                    .originStatsFactory(originStatsFactory)
+                    .connectionPoolFactory(connectionPoolFactory)
+                    .initialOrigins(backendService.origins())
                     .build();
             return new ProxyToBackend(clientFactory.createClient(backendService, inventory));
         }
+
+        private ConnectionPoolFactory connectionPoolFactory(
```

Comment:
This method can be static.

Reply:
This method is just a wrapper around `new` keyword. I will remove it.